instrumentation/kafka-clients/src/main/java/brave/kafka/TracingConsumer.java

```diff
@@ -0,0 +1,193 @@
+package brave.kafka;
+
+import brave.Tracing;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.consumer.OffsetAndTimestamp;
+import org.apache.kafka.clients.consumer.OffsetCommitCallback;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+
+/**
+ * Kafka Consumer decorator. Read records headers and finish producers spans if possible.
+ */
+class TracingConsumer<K, V> implements Consumer<K, V> {
+
+  private final RecordTracing recordTracing;
+  private Consumer<K, V> wrappedConsumer;
+
+  TracingConsumer(Tracing tracing, Consumer<K, V> consumer) {
```

Comment:
Not sure to understand.
Do you want me to add another constructor with KafkaTracing ? That seems strange as KafkaTracing is responsible of the TracingConsumer creation.